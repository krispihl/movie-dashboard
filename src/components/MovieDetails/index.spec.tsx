import { MovieDetails } from "./index";
import { render, screen } from '@testing-library/react';

jest.mock('../../contexts/movie-details', () => ({
    useMovieDetails: () => ({
        showMovieDetails: true,
        setBackDropSpacing: jest.fn(),
    })
}));

jest.mock('../../contexts/movies', () => ({
    useMovies: () => ({
        genres: [
            {
                id: 28,
                name: 'Action',
                popular: false,
            },
            {
                id: 12,
                name: 'Adventure',
                popular: false,
            },
            {
                id: 16,
                name: 'Animation',
                popular: true,
            },
        ],
    })
}));

jest.mock('../Rating', () => ({
    Rating: () => {
        return <div className='mocked-rating' />
    }
}));

const mockedData = {
    adult: false,
    backdrop_path: '/backdrop.jpg',
    genre_ids: [16, 28],
    id: 123456,
    original_language: 'en',
    original_title: 'The Movie',
    overview: 'Story about writing tests',
    popularity: 500,
    poster_path: '/poster.jpg',
    release_date: '2023-03-07',
    title: 'The Movie',
    video: false,
    vote_average: 8.0,
    vote_count: 666,
    rowPosition: 0,
};

const mockedConfig = {
    base_url: 'http://image.tmdb.org/t/p/',
    secure_base_url: 'https://image.tmdb.org/t/p/',
    backdrop_sizes: [],
    logo_sizes: [],
    poster_sizes: [],
    profile_sizes: [],
    still_sizes: [],
};

describe('src/components/MovieDetails', () => {
    it('should render MovieDetails correctly', async () => {
        const { container } = render(<MovieDetails data={mockedData} config={mockedConfig} />);

        expect(container).toMatchSnapshot();
        expect(screen.getByRole('heading')).toHaveTextContent('The Movie');
        expect(document.querySelectorAll('li')).toHaveLength(2);
        expect(document.querySelector("[alt='The Movie']")).toBeInTheDocument();
    });
});
