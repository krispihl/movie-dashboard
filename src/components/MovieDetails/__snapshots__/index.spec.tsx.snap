// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/MovieDetails should render MovieDetails correctly 1`] = `
<div>
  <div
    class="details__pointer"
  />
  <div
    class="details__pointer--border"
  />
  <div
    class="details"
    style="left: 0px;"
  >
    <div
      class="details__data"
      id="back-123456"
    >
      <h2
        class="details__data--title"
      >
        The Movie
      </h2>
      <div
        class="mocked-rating"
      />
      <p
        class="details__data--overview"
      >
        Story about writing tests
      </p>
      <ul
        class="details__data--genres"
      >
        <li>
          #Animation
        </li>
        <li>
          #Action
        </li>
      </ul>
    </div>
    <div
      class="details__image-wrapper"
    >
      <img
        alt="The Movie"
        src="http://image.tmdb.org/t/p/w780/backdrop.jpg"
        style="height: 0px;"
      />
    </div>
  </div>
</div>
`;
